File: Programmers/구명보트/Hong/people.cpp
#include<iostream>
#include<vector>
#include<algorithm>
#include<queue>
#include<string>
#include<set>
#include<map>
#include<cmath>
#include<stack>
#include<utility>

using namespace std;

int solution(vector<int> people, int limit) {
	int answer = 0;
	int right = people.size() - 1;
	int left = 0;
	sort(people.begin(), people.end());

	while (right >= left)
	{
		if (limit >= people[right] + people[left])
		{
			answer++;
			right--;
			left++;
		}
		else
		{
			right--;
			answer++;
		}
	}

	return answer;
}

int main()
{
	vector<int> people = { 70, 50, 80, 50 };
	int limit = 100;

	cout << solution(people, limit);
}

File: Programmers/구명보트/Hong/people.java
package com.company;

import java.util.*;

class Solution {
    public int solution(int[] people, int limit) {
        int answer = 0;
        int size = people.length;
        int left = 0;
        int right = size - 1;
        Arrays.sort(people);

        while(left <= right){
            if(left == right){
                answer++;
                break;
            }

            if(people[left] + people[right] <= limit){
                answer++;
                left++;
                right--;
                continue;
            }
            else{
                answer++;
                right--;
            }
        }
        return answer;
    }
}

class Main {

    public static void main(String[] args) {
        int[] people = {50,70,80}; //50 50 70 80
        int limit = 100;

        Solution solution = new Solution();
        System.out.println(solution.solution(people,limit));
    }
}